### PR TITLE
feat(style)!: refactor how to build the overlay style

### PR DESCRIPTION
**Pull request by csouchet**
To address issue #13 efficiently and avoid creating separate functions for creating `font`, `fill` and `stroke` styles for `overlay`, `edge` and `shape`, I refactored the existing code.

As part of this refactoring, I've converted the `create_font`, `create_fill` and `create_stroke` functions to internal functions.  
This enhancement allows these functions to accept additional parameters as needed. The responsibility for passing the appropriate parameters to these functions now rests with the functions dedicated to overlays (`create_overlay_style`), edges, and shapes.  
 

BREAKING CHANGE: `create_font`, `create_fill` and `create_stroke` functions are internal functions. `create_style` has been renamed in `create_overlay_style`, and now, we use only this function to create the style of an overlay.

### CODE REVIEW

**Review by tbouffard — COMMENTED** [review]
⚠️ **This PR introduces breaking changes**, so its title and content should mention it. I suggest something like `refactor!: simplify the construction of the overlay style` for the PR title.
The commit used to merge the PR must also mention it and the release notes as well.
See https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-both--and-breaking-change-footer

I don't know how breaking changes have to be handled in the R ecosystem. This is something that should be checked IMHO.

I don't see any problem for introducing breaking changes in general. Can you please give me an example of the future "style update" functions to let me understand how the refactoring help for the new API?
In parallel, I am reviewing the details of the code changes.

File: R/funs.R [hunk 1]
```diff
@@ -153,43 +172,41 @@ create_font <- function(color = NULL, size = NULL) {
...
+#' @noRd
 create_fill <- function(color) {
   ret <-
     .not_null_list(
       color = color
     )
 }
 
-#' @title Create the stroke style of an overlay
+#' @title Internal fun to create the stroke style of an overlay or a 'BPMN' element
```
Comment (tbouffard) [inline R/funs.R:196]: **suggestion:** 
```suggestion
#' @title Internal function to create the stroke style of an overlay or a 'BPMN' element
```

File: R/funs.R [hunk 2]
```diff
@@ -17,7 +17,17 @@
 #' Use these constants as the \code{position} argument in the \code{\link{create_overlay}} function.
 #'
 #' @export
-overlay_shape_position <- c("top-left", "top-right", "top-center", "bottom-left", "bottom-right", "bottom-center", "middle-left", "middle-right")
+overlay_shape_position <-
```
Comment (tbouffard) [inline R/funs.R:20]: **suggestion:** I have noticed an rendering issue in the documentation with the list of `overlay_shape_position` and for `overlay_edg_position`. The issue was already there.
We may fix it as part of this PR to further improve the documentation.

image
_`overlay_shape_position` rendering issue_

Comment (csouchet) [reply R/funs.R:29]: ### After fix

File: R/funs.R [hunk 3]
```diff
@@ -102,49 +118,52 @@ create_overlay <- function(elementId, label, style = NULL, position = NULL) {
...
 #'
 #' Use this function to create the correct style structure for an overlay.
 #'
-#' @param font The font style of the overlay
-#'      Use \code{\link{create_font}} function to create the font style object for the overlay.
-#' @param fill The fill style of the overlay
-#'      Use \code{\link{create_fill}} function to create the fill style object for the overlay.
-#' @param stroke The stroke style of the overlay
-#'      Use \code{\link{create_stroke}} function to create the stroke style object for the overlay.
+#' @param font_color The font color of the overlay. Use a custom color string.
```
[the remark this answers is not in the document]
Comment (tbouffard) [reply R/funs.R]: OK, I probably confused it with the _Update Stye_ API in  that provides more information: https://process-analytics.github.io/bpmn-visualization-js/api/types/Font.html 😄 
"Use a custom color string." was unclear to me. I suggest to remove it.

File: R/funs.R (hunk 2 above)
Comment (tbouffard) [reply R/funs.R:29]: Nice!

File: R/funs.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (tbouffard) [reply R/funs.R]: The comment is now "Use all HTML color names or HEX codes." which is great. So resolving this comment.

**Pull request merged**